Model/PayolutionInvoicePaymentMethod.php

```diff
@@ -107,8 +129,95 @@ public function getConfigFields()
+            ],
+            'apiUrl' => [
+                'type' => 'text',
+                'field' => 'oxpayments__wdoxidee_apiurl',
+                'title' => Helper::translate('wd_config_base_url'),
+                'description' => Helper::translate('wd_config_base_url_desc'),
+                'required' => true,
+            ],
         ];
-        return parent::getConfigFields() + $aAdditionalFields;
+        return $aConfigFields + $this->_getCustomCurrencyConfigFields();
+    }
+
+    /**
+     * Returns an array of all custom config fields per activated currency.
+     * One separator, HTTP user/password and MAID and secret fields are added per currency.
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    private function _getCustomCurrencyConfigFields()
+    {
+        $aCurrencyFields = [];
+
+        $aCurrencies = $this->_oPayment->oxpayments__allowed_currencies->value;
+
+        // fields that are configurable per currency
+        $aFields = [
+            'httpUser' => [
+                'fieldType' => 'text',
+                'dbFieldPrefix' => 'oxpayments__httpuser',
+                'title' => Helper::translate('wd_config_http_user'),
+            ],
+            'httpPassword' => [
+                'fieldType' => 'text',
+                'dbFieldPrefix' => 'oxpayments__httppass',
+                'title' => Helper::translate('wd_config_http_password'),
+            ],
+            'maid' => [
+                'fieldType' => 'text',
+                'dbFieldPrefix' => 'oxpayments__maid',
+                'title' => Helper::translate('wd_config_merchant_account_id'),
+                'description' => Helper::translate('wd_config_merchant_account_id_desc'),
+            ],
+            'secret' => [
+                'fieldType' => 'text',
+                'dbFieldPrefix' => 'oxpayments__secret',
+                'title' => Helper::translate('wd_config_merchant_secret'),
+                'description' => Helper::translate('wd_config_merchant_secret_desc'),
+            ],
+         ];
+
+        foreach ($aCurrencies as $sCurrency) {
+            $aCurrencyFields['groupSeparator_' . $sCurrency] = [
+                'type' => 'separator',
+                'title' => $sCurrency,
+            ];
+
+            foreach ($aFields as $sFieldName => $aConfigProps) {
+                $sConfigKey = $sFieldName . '_' . $sCurrency;
+                $aFieldProps = [
+                    'type' => $aConfigProps['fieldType'],
+                    'field' => $aConfigProps['dbFieldPrefix'] . '_' . strtolower($sCurrency),
+                    'title' => $aConfigProps['title'],
+                ];
+
+                if (isset($aConfigProps['description'])) {
+                    $aFieldProps['description'] = $aConfigProps['description'];
+                }
+
+                $aCurrencyFields[$sConfigKey] = $aFieldProps;
+            }
+
+            $aCurrencyFields['testCredentials_' . $sCurrency] = [
```

Comment:
If you go with the string replacement suggestion above you can also define this field in the "field templates":

```php
'testCredentials_%s' => [
    'type' => 'button',
    'onclick' => "wdTestPaymentMethodCredentials('%s')",
    'text' => Helper::translate('wd_test_credentials'),
    'colspan' => '2',
]
```

Reply:
testCredentials field has different keys than other fields, thus making its integration in for loop a bit complicated.